hypervisor/xen/xen.go

```diff
@@ -244,38 +244,37 @@ func (xc *XenContext) AddNic(ctx *hypervisor.VmContext, host *hypervisor.HostNic
 
 		glog.V(1).Infof("allocate nic %s for dom %d", host.Mac, xc.domId)
 		hw, err := net.ParseMAC(host.Mac)
-		if err == nil {
-			dev := host.Device
-			glog.V(1).Infof("add network for %d - ip: %s, br: %s, gw: %s, dev: %s, hw: %s", xc.domId, guest.Ipaddr,
-				host.Bridge, host.Bridge, dev, hw.String())
-
-			res := HyperxlNicAdd(xc.driver.Ctx, (uint32)(xc.domId), guest.Ipaddr, host.Bridge, host.Bridge, dev, []byte(hw))
-			if res == 0 {
-
-				glog.V(1).Infof("nic %s insert succeeded", guest.Device)
-
-				err = network.UpAndAddToBridge(dev, "", "")
-				if err != nil {
-					glog.Error("fail to add vif to bridge: ", err.Error())
-					ctx.Hub <- &hypervisor.DeviceFailed{
-						Session: callback,
-					}
-					HyperxlNicRemove(xc.driver.Ctx, (uint32)(xc.domId), host.Mac)
-					return
-				}
-
-				result <- callback
-				return
+		if err != nil {
+			glog.Errorf("failed to parse MAC(%s): %v", host.MAC, err)
+			result <- &hypervisor.DeviceFailed{
+				Session: callback,
 			}
+			return
+		}
+
+		dev := host.Device
+		glog.V(1).Infof("add network for %d - ip: %s, br: %s, gw: %s, dev: %s, hw: %s", xc.domId, guest.Ipaddr,
+			host.Bridge, host.Bridge, dev, hw.String())
+
+		res := HyperxlNicAdd(xc.driver.Ctx, (uint32)(xc.domId), guest.Ipaddr, host.Bridge, host.Bridge, dev, []byte(hw))
+		if res != 0 {
 			glog.V(1).Infof("nic %s insert succeeded [faked] ", guest.Device)
 			result <- callback
 			return
 		}
 
-		glog.Errorf("nic %s insert failed", guest.Device)
```

Comment:
I do not think this log is reasonable. 
It should be `glog.Errorf("failed to parse MAC(%s): %v", host.MAC, err)`. So I added this in the upper area of this function.